src/Microsoft.AspNetCore.Cors/Infrastructure/CorsMiddleware.cs

```diff
@@ -85,11 +76,11 @@ public class CorsMiddleware
         }
 
         /// <inheritdoc />
-        public async Task Invoke(HttpContext context)
+        public async Task Invoke(HttpContext context, ICorsPolicyProvider corsPolicyProvider)
         {
             if (context.Request.Headers.ContainsKey(CorsConstants.Origin))
             {
-                var corsPolicy = _policy ?? await _corsPolicyProvider?.GetPolicyAsync(context, _corsPolicyName);
+                var corsPolicy = _policy ?? await corsPolicyProvider?.GetPolicyAsync(context, _corsPolicyName);
```

Comment:
Pretty sure this can't be null.

Reply:
You're right. Invoking middleware will fail if it can't resolve the parameters, right? I'll remove the `?`.